lm75bd/lm75bd.c

```diff
@@ -25,9 +26,43 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
   return ERR_CODE_SUCCESS;
 }
 
+// 2 Bytes of Data
+#define CONF_POINTER_BUFF_SIZE 1U
+#define CONF_READ_BUFF_SIZE 2U
+#define MASK_SIGN_BIT 0x7FU
+#define MASK_NINE_BIT 0x1FFU
+/**
+ * @brief Reads LM75BD temperature using I2C
+ */
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
-  
+  error_code_t errCode;
+
+  /* Write pointer register */
+  uint8_t point_buff[CONF_POINTER_BUFF_SIZE] = {0};
+  point_buff[0] = 0x00U;
+ 
+  errCode = i2cSendTo(LM75BD_OBC_I2C_ADDR, point_buff, CONF_POINTER_BUFF_SIZE);
+  if (errCode != ERR_CODE_SUCCESS)
+    return errCode;
+
+  /* Read temperature register */
+  uint8_t read_buff[CONF_READ_BUFF_SIZE] = {0};
+  errCode = i2cReceiveFrom(LM75BD_OBC_I2C_ADDR, read_buff, CONF_READ_BUFF_SIZE);
+  if (errCode != ERR_CODE_SUCCESS)
+    return errCode;
```

Comment:
use `RETURN_IF_ERROR_CODE` to make sure this error gets logged as well

Reply:
Added null checks